src/tests/ftest/launch.py

```diff
@@ -20,164 +20,549 @@
+
+    Returns:
+        dict: a dictionary of DAOS file name keys and full path values
+
+    """
+    # List of default DAOS files
+    if daos_files is None:
+        daos_files = {
+            "log_file": "/tmp/server.log",
+            "agent_log_file": "/tmp/daos_agent.log",
+            "control_log_file": "/tmp/daos_control.log",
+            "socket_dir": "/tmp/daos_sockets",
+            "daos_log_file": "/tmp/daos.log",
+        }
+
+    # Determine the log file locations defined by the last run test
+    print("Checking {} for daos log file locations".format(config_yaml))
+    yaml_data = get_yaml_data(config_yaml)
+
+    # Replace any default log file with its yaml definition
+    matches = find_values(yaml_data, daos_files.keys(), val_type=str)
+    for key, value in matches.items():
+        if value != daos_files[key]:
+            print(
+                "  Update found for {}: {} -> {}".format(
+                    key, daos_files[key], value))
+            daos_files[key] = value
+
+    return daos_files
+
+
+def get_hosts_from_yaml(test_yaml):
+    """Extract the list of hosts from the test yaml file.
+
+    This host will be included in the list if no clients are explicitly called
+    out in the test's yaml file.
+
+    Args:
+        test_yaml (str): test yaml file
+
+    Returns:
+        list: a unique list of hosts specified in the test's yaml file
+
+    """
+    yaml_data = get_yaml_data(test_yaml)
+    server_keys = [
+        "test_machines",
+        "test_servers",
+        "daos_servers",
+        "servers",
+        "test_machines1",
+        "test_machines2",
+        "test_machines2a",
+        "test_machines3",
+        "test_machines6"]
+    client_keys = [
+        "test_clients",
+        "clients",
+    ]
+    host_set = set()
+    found_client_key = False
+    matches = find_values(yaml_data, server_keys + client_keys)
+    for key, value in matches.items():
+        host_set.update(value)
+        if key in client_keys:
+            found_client_key = True
+
+    # Include this host as a client if no clients are specified
+    if not found_client_key:
+        host_set.add(socket.gethostname().split(".")[0])
+
+    return sorted(list(host_set))
+
+
+def clean_logs(test_yaml):
+    """Remove the test log files on each test host.
+
+    Args:
+        test_yaml (str): yaml file containing host names
+    """
+    # Use the default server yaml and then the test yaml to update the default
+    # DAOS log file locations.  This should simulate how the test defines which
+    # log files it will use when it is run.
+    log_files = get_log_files(test_yaml, get_log_files(BASE_LOG_FILE_YAML))
+    host_list = get_hosts_from_yaml(test_yaml)
+    command = "ssh {{host}} \"rm -fr {}\"".format(" ".join(log_files.values()))
+    print("Cleaning logs on {}".format(host_list))
+    spawn_commands(host_list, command)
+
+
+def archive_logs(avocado_logs_dir, test_yaml):
```

Comment:
Is https://github.com/daos-stack/daos/blob/e320c1163582632738886c5bed72c0d9a4c3661d/ftest.sh#L380-L392 still needed with this now?

Reply:
I will remove it.